### PR TITLE
Avoid Array.ofDim where possible 

### PR DESCRIPTION
**Pull request by johanandren**
Refs #22516

### CODE REVIEW

File: akka-actor/src/main/scala/akka/util/ByteIterator.scala [hunk 1]
```diff
@@ -13,15 +13,14 @@ import scala.reflect.ClassTag
...
-    private val emptyArray: Array[Byte] = Array.ofDim[Byte](0)
 
     protected[akka] def apply(array: Array[Byte]): ByteArrayIterator =
       new ByteArrayIterator(array, 0, array.length)
 
     protected[akka] def apply(array: Array[Byte], from: Int, until: Int): ByteArrayIterator =
       new ByteArrayIterator(array, from, until)
 
-    val empty: ByteArrayIterator = apply(emptyArray)
+    val empty: ByteArrayIterator = apply(Array.emptyByteArray)
```
Comment (johanandren) [inline akka-actor/src/main/scala/akka/util/ByteIterator.scala]: Existing constant in the standard library!

**Review by patriknw — APPROVED** [review]
LGTM

**Review by 2m — APPROVED** [review]
LGTM. Do you think it would be worth to setup https://scalacenter.github.io/scalafix/ in order to ensure that `new Array` is always used instead of `Array.ofDim`?

**Comment by johanandren** [comment]
No, because it isn't _always_ used, only when we know the concrete type at the instantiation site.

**Pull request merged**